src/ir/effects.h

```diff
@@ -770,6 +770,19 @@ class EffectAnalyzer {
     void visitStringEncode(StringEncode* curr) {
       // traps when ref is null or we write out of bounds.
       parent.implicitTrap = true;
+      switch (curr->op) {
+        case StringEncodeUTF8:
+        case StringEncodeWTF8:
+        case StringEncodeWTF16:
+          parent.writesMemory = true;
+          break;
+        case StringEncodeUTF8Array:
+        case StringEncodeWTF8Array:
+        case StringEncodeWTF16Array:
+          parent.writesArray = true;
+          break;
+        default: {}
```

Comment:
I guess need also add `parent.readsMemory = true` for StringMeasureOp's like `StringMeasureUTF8`...`StringMeasureWTF16View` and `StringSliceWTFOp`

Reply:
Actually not StringMeasure, which operates on a String, but StringNew does need additions here.